src/eHealth.locations/services/selected-location-factory.js

```diff
@@ -40,19 +40,7 @@ angular.module('eHealth.locations.services')
               items: original.items,
               original: original
             };
-        if (hasAllItem) {
-          // adds an all-item.
-          // The all-item will be automatically selected when there is not
-          // a more narrow selection for this level or a child level
-          var allItem = {
-            isAll: true,
-            name: 'all'
-          };
-          // do not edit `level.items` in place! it might refer to
-          // external location data which we don't want to mess up
-          level.items = original.items.concat(allItem);
-          level.selected = allItem;
-        }
+        removeFilters(true);
```

Comment:
The first law for laziness says: "Don't do it twice if you can do it just once" :construction_worker: